carp.clients.core/src/commonMain/kotlin/dk/cachet/carp/clients/domain/StudyRuntime.kt

```diff
@@ -191,7 +192,10 @@ class StudyRuntime private constructor(
                     ?: throw UnsupportedOperationException( "Connecting to device of type \"$deviceType\" is not supported on this client." )
             }
 
-            val dataTypes = tasks.flatMap { it.measures }.map { it.type }.distinct()
+            val dataTypes = tasks
+                .flatMap { it.measures.filterIsInstance<Measure.DataStream>() }
```

Comment:
I wonder if task measures should be split up rather than "overloaded" like this.

Reply:
Both are measures. The other instance is simply `Measure.TriggerData` which should _always_ supported, and `Measure.TriggerData` does not contain a `dataType` (since it is _always_ the same data type). The difference is that for `Measure.DataStream` the researcher could incorrectly configure a `dataType` to be measured on a device, which is not supported at runtime.